prow/cmd/pipeline/main.go

```diff
@@ -150,6 +152,11 @@ func main() {
 
 	pipelineConfigs := map[string]pipelineConfig{}
 	for context, cfg := range configs {
+		// Don't panic when a build cluster cannot be reached
```

Comment:
The client creation runs a full api discovery which is an expensive way to check if a cluster is reachable. Wouldn't it be simpler to instead replace the `if err != nil { Fatalf()}` statement below with a Warning and a `continue`?

Reply:
Agreed and I like your approach better. Would like to hear what's @fejta think about this approach

Reply:
sgtm

Though in an ideal world a transient error when starting the pipeline controller would not cause this context to be permanently unavailable. AKA we'd retry every minute or 10 or whatever and add it to the map once available -- but that'll be a more involved change.

Reply:
Done